src/test-utilities/legacy.tsx

```diff
@@ -76,38 +73,10 @@ function updateRoot(wrapper: AnyWrapper) {
   (wrapper as any).root().update();
 }
 
-type AppContext = {
-  intl: I18n;
-  scrollLockManager: ScrollLockManager;
-  stickyManager: StickyManager;
-  appBridge: ClientApplication<{}> | {} | null;
-  themeProvider: ThemeProviderContextType;
-  frame: FrameContextType;
-  link: Link;
-};
-
-interface AppContextOptions {
-  app: AppContext;
-}
-
-interface MountWithAppProviderOptions {
-  context?: {
-    themeProvider?: DeepPartial<ThemeProviderContextType>;
-    frame?: DeepPartial<FrameContextType>;
-    intl?: TranslationDictionary | TranslationDictionary[];
-    scrollLockManager?: ScrollLockManager;
-    stickyManager?: StickyManager;
-    appBridge?: ClientApplication<{}> | {};
-    link?: LinkLikeComponent;
-  };
-}
-
 export function mountWithAppProvider<P>(
   node: React.ReactElement<P>,
-  options: MountWithAppProviderOptions = {},
-): PolarisContextReactWrapper<P, any> {
-  const {context: ctx = {}} = options;
-
+  ctx: WithProvidersOptions = {},
```

Comment:
Just a thought if we decide to change `WithProvidersOptions` type in the future, that'll require some manual updates in legacy tests. I'd like us to be in a spot where we'll never have to touch legacy stuff again (unless we need to fix a test)

Reply:
If we were to ever add a new context into AppProvider, we'd want to add it to both the legacy and modern test harnesses, having to duplicate that feels awkward.

I think the end goal of this is to make our mountWithAppProvider and mountWithContext functions a lot slimmer and instead have all the logic around providing defaults values for contexts done in PolarisTextProvider.